File: app/src/main/java/com/uncmorfi/helpers/ParserHelper.java
package com.uncmorfi.helpers;

import com.github.mikephil.charting.data.Entry;
import com.uncmorfi.menu.DayMenu;

import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Comparator;
import java.util.Date;
import java.util.Locale;
import java.util.TimeZone;

public abstract class ParserHelper {

    public static Date stringToDate(String string) {
        final DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.getDefault());
        dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
        try {
            return dateFormat.parse(string);
        } catch (ParseException e) {
            e.printStackTrace();
            return null;
        }
    }

    public static long clearDate(Date date) {
        Calendar cal = Calendar.getInstance();
        cal.setTime(date);
        cal.set(Calendar.YEAR, 1970);
        cal.set(Calendar.MONTH, 0);
        cal.set(Calendar.DAY_OF_MONTH, 1);
        return cal.getTimeInMillis()/1000;
    }

    public static class MenuDayComparator implements Comparator<DayMenu>
    {
        public int compare(DayMenu left, DayMenu right) {
            return left.getDate().compareTo(right.getDate());
        }
    }

    public static class CounterEntryComparator implements Comparator<Entry>
    {
        public int compare(Entry left, Entry right) {
            return new Date((long) left.getX()).compareTo(new Date((long)right.getX()));
        }
    }
}

File: app/src/main/java/com/uncmorfi/helpers/ParserHelper.kt
package com.uncmorfi.helpers

import com.uncmorfi.models.DayMenu
import com.uncmorfi.models.Serving
import org.json.JSONArray
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

fun JSONArray.toArray() : Array<String> {
    val result = ArrayList<String>()
    for (i in 0 until this.length()) {
        result.add(this.getString(i))
    }
    return result.toTypedArray()
}

fun String.toDate(timeZone: String? = null): Date? {
    val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.getDefault())
    timeZone?.let { dateFormat.timeZone = TimeZone.getTimeZone(it) }
    val date: Date
    try {
        date = dateFormat.parse(this)
    } catch (e: ParseException) {
        e.printStackTrace()
        return null
    }

    return date
}

fun Date.clearDate(): Long {
    val cal = Calendar.getInstance()
    cal.time = this
    cal.set(Calendar.YEAR, 1970)
    cal.set(Calendar.MONTH, 0)
    cal.set(Calendar.DAY_OF_MONTH, 1)
    return cal.timeInMillis / 1000
}

fun Date.toFormat(format: String) : String {
    val fmt = SimpleDateFormat(format, Locale.getDefault())
    return fmt.format(this)
}

fun Date.compareToToday(): Int {
    val fmt = SimpleDateFormat("yyyyMMdd", Locale.getDefault())
    return fmt.format(this).compareTo(fmt.format(Date()))
}

object ParserHelper {
    class MenuDayComparator : Comparator<DayMenu> {
        override fun compare(left: DayMenu, right: DayMenu): Int {
            return left.date.compareTo(right.date)
        }
    }

    class ServingsComparator : Comparator<Serving> {
        override fun compare(left: Serving, right: Serving): Int {
            return left.date.compareTo(right.date)
        }
    }
}